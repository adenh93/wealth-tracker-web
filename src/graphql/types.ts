export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type CryptoCurrency = {
  __typename?: 'CryptoCurrency';
  id: Scalars['Int'];
  name: Scalars['String'];
  rank: Scalars['Int'];
  symbol: Scalars['String'];
};

export type CryptoCurrencyHolding = {
  __typename?: 'CryptoCurrencyHolding';
  cryptoCurrency?: Maybe<CryptoCurrency>;
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
  id: Scalars['Int'];
  netHoldingsValue: Scalars['Float'];
  percentChange24h: Scalars['Float'];
  price: Scalars['Float'];
};

export type CryptoCurrencyHoldingsAddInput = {
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
};

export type CryptoCurrencyHoldingsUpdateInput = {
  holdings: Scalars['Float'];
  id: Scalars['Int'];
};

export type Mutation = {
  __typename?: 'Mutation';
  cryptoCurrencyHoldingsAdd?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsDelete?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsUpdate?: Maybe<CryptoCurrencyHolding>;
};


export type MutationCryptoCurrencyHoldingsAddArgs = {
  input: CryptoCurrencyHoldingsAddInput;
};


export type MutationCryptoCurrencyHoldingsDeleteArgs = {
  id: Scalars['Int'];
};


export type MutationCryptoCurrencyHoldingsUpdateArgs = {
  input: CryptoCurrencyHoldingsUpdateInput;
};

export type Query = {
  __typename?: 'Query';
  cryptoCurrencies?: Maybe<Array<CryptoCurrency>>;
  cryptoCurrencyHoldings?: Maybe<Array<CryptoCurrencyHolding>>;
};


export type QueryCryptoCurrenciesArgs = {
  query?: Maybe<Scalars['String']>;
};
