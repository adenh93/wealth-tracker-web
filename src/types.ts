export interface Asset {
  id: string
  logoSrc: string
  name: string
  ticker: string
  price: number
  gain24Hr: number
  holdings: number
}
