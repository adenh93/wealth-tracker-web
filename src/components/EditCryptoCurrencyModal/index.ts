export { default } from './EditCryptoCurrencyModal'
export type { EditCryptoCurrencyModalProps } from './EditCryptoCurrencyModal'
