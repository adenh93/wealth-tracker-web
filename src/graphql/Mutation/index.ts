export { default as CRYPTOCURRENCY_HOLDINGS_ADD } from './cryptoCurrencyHoldingsAdd'
export { default as CRYPTOCURRENCY_HOLDINGS_UPDATE } from './cryptoCurrencyHoldingsUpdate'
export { default as CRYPTOCURRENCY_HOLDINGS_DELETE } from './cryptoCurrencyHoldingsDelete'
