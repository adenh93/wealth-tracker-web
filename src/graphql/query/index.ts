export { default as CRYPTO_CURRENCIES } from './cryptoCurrencies'
