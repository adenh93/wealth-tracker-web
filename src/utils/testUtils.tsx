import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
