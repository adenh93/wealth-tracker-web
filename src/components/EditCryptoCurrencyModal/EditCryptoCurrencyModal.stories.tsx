import { Story } from '@storybook/react'
import EditCryptoCurrencyModal, {
  EditCryptoCurrencyModalProps,
} from './EditCryptoCurrencyModal'
import { CryptoCurrencyHolding } from '../../graphql/types'

const config = {
  title: 'Components/EditCryptoCurrencyModal',
  component: EditCryptoCurrencyModal,
}

const Template: Story<EditCryptoCurrencyModalProps> = (args) => (
  <EditCryptoCurrencyModal {...args} />
)

export const Primary = Template.bind({})

const asset: CryptoCurrencyHolding = {
  cryptoCurrency: {
    id: 1,
    rank: 1,
    name: 'Bitcoin',
    symbol: 'BTC',
  },
  id: 2,
  cryptoCurrencyId: 1,
  holdings: 15.25,
  price: 80532.5,
  percentChange24h: 1.54,
  netHoldingsValue: 1228120.625,
}

Primary.args = {
  asset,
  open: true,
  errors: {},
  handleSubmit: () => {},
  handleClose: () => {},
}

export default config
