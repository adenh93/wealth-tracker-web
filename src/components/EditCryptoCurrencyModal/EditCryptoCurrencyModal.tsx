import { FC } from 'react'
import { TextField, Typography } from '@mui/material'
import { useForm } from 'react-hook-form'
import ConfirmModal from '../ui/ConfirmModal'
import { CryptoCurrencyHolding } from '../../graphql/types'

export interface EditCryptoCurrencyModalProps {
  asset: CryptoCurrencyHolding
  open: boolean
  handleSave: () => void
  handleClose: (e: any) => void
}

const EditCryptoCurrencyModal: FC<EditCryptoCurrencyModalProps> = ({
  asset,
  open,
  handleSave,
  handleClose,
}) => {
  const { register, handleSubmit } = useForm()

  return (
    <ConfirmModal
      open={open}
      title="Edit Asset"
      handleConfirm={handleSubmit(handleSave)}
      handleClose={handleClose}
    >
      <Typography variant="body2" sx={{ mb: 2 }}>
        Editing holdings for {asset.cryptoCurrency?.name} (
        {asset.cryptoCurrency?.symbol}):
      </Typography>
      <TextField
        {...register('newHoldings')}
        variant="standard"
        label="New Holdings"
        type="number"
        data-testid="new-holdings-input"
      />
    </ConfirmModal>
  )
}

export default EditCryptoCurrencyModal
