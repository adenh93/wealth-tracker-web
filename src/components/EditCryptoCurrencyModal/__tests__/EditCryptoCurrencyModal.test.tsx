import { CryptoCurrencyHolding } from '../../../graphql/types'
import { render } from '../../../utils/testUtils'
import EditCryptoCurrencyModal from '../EditCryptoCurrencyModal'

describe('EditCryptoCurrencyModal component tests', () => {
  const mockAsset: CryptoCurrencyHolding = {
    cryptoCurrency: {
      id: 1,
      rank: 1,
      name: 'Bitcoin',
      symbol: 'BTC',
    },
    id: 2,
    cryptoCurrencyId: 1,
    holdings: 15.25,
    price: 80532.5,
    percentChange24h: 1.54,
    netHoldingsValue: 1228120.625,
  }

  test('matches snapshot', () => {
    const { asFragment } = render(
      <EditCryptoCurrencyModal
        asset={mockAsset}
        open={true}
        errors={{}}
        submitting={false}
        register={jest.fn()}
        handleSubmit={jest.fn()}
        handleClose={jest.fn()}
      />
    )

    expect(asFragment).toMatchSnapshot()
  })

  test('renders asset name correctly', () => {
    const { getByText } = render(
      <EditCryptoCurrencyModal
        asset={mockAsset}
        open={true}
        errors={{}}
        submitting={false}
        register={jest.fn()}
        handleSubmit={jest.fn()}
        handleClose={jest.fn()}
      />
    )

    expect(getByText('Editing holdings for Bitcoin (BTC):')).not.toBeNull()
  })
})
