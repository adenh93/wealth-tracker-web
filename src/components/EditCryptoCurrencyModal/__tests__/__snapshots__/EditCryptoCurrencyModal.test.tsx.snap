// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCryptoCurrencyModal component tests matches snapshot 1`] = `[Function]`;
